### PR TITLE
Feature/slim 2863 retry on all exceptions

### PR DESCRIPTION
**Pull request by jboon**
https://smartsocietyservices.atlassian.net/browse/SLIM-2863

### CODE REVIEW

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.RetryHeaderFactory;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.BundleMessagesRequestDto;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
```
Comment (JelleHoffman) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java:47]: Why is this line necessary? Have you added unused mock elements by any change? It would be nicer without this line of code.

**Review by JelleHoffman — CHANGES_REQUESTED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java (hunk 1 above)
Comment (JelleHoffman) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java:47]: However it is not a must.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    @Mock
+    private DlmsConnectionManager dlmsConnectionManager;
+
+    @Mock
+    private DlmsMessageListener messageListener;
+
+    @InjectMocks
+    private BundleMessageProcessor messageProcessor;
+
+    private DlmsDevice dlmsDevice = new DlmsDeviceBuilder().withHls5Active(true).build();
```
Comment (coendamen) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java:83]: minor: maybe put this in the setUp method so all init stuff is in the same place.

**Review by coendamen — APPROVED** [review]

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java (hunk 1 above)
Comment (jboon) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java:47]: No, good find, I forgot to remove this.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java (hunk 2 above)
Comment (jboon) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java:83]: yes

**Review by JelleHoffman — APPROVED** [review]
